core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -242,11 +248,16 @@ public IndexShard(ShardId shardId, @IndexSettings Settings indexSettings, ShardP
         this.flushThresholdSize = indexSettings.getAsBytesSize(INDEX_TRANSLOG_FLUSH_THRESHOLD_SIZE, new ByteSizeValue(512, ByteSizeUnit.MB));
         this.disableFlush = indexSettings.getAsBoolean(INDEX_TRANSLOG_DISABLE_FLUSH, false);
         this.indexShardOperationCounter = new IndexShardOperationCounter(logger, shardId);
+        this.indexingMemoryController = provider.getIndexingMemoryController();
+
         this.searcherWrapper = provider.getIndexSearcherWrapper();
         this.percolatorQueriesRegistry = new PercolatorQueriesRegistry(shardId, indexSettings, queryParserService, indexingService, mapperService, indexFieldDataService);
         if (mapperService.hasMapping(PercolatorService.TYPE_NAME)) {
             percolatorQueriesRegistry.enableRealTimePercolator();
         }
+
+        // TODO: can we somehow call IMC.forceCheck here?  Since we just became active, it can divvy up the RAM
```

Comment:
OK I added a line at the end of `IndexService.createShard` to invoke `forceCheck` ... but are we worried about the O(N^2) cost here?  N should be smallish (number of nodes on the shard)?